sqlconnect/internal/snowflake/config.go

```diff
@@ -34,11 +34,19 @@ type Config struct {
 	KeepSessionAlive  bool   `json:"keepSessionAlive"`
 	UseLegacyMappings bool   `json:"useLegacyMappings"`
 	QueryTag          string `json:"queryTag"`
+	Host               string `json:"host"`
+	Token              string `json:"token"`
+	Authenticator      gosnowflake.AuthType `json:"authenticator"`
```

Comment:
What other type of authenticators do we need to support? Would it be possible to run all scenarios for this new auth method [here](https://github.com/rudderlabs/sqlconnect-go/blob/main/sqlconnect/internal/snowflake/integration_test.go#L21) too?

Reply:
Following the existing `UseKeyPairAuth` pattern, we could do the following for more predictable results:
```
UseOAuth bool
OAuthToken string
```

Reply:
Updated